### PR TITLE
Add job timeout to builder

### PR DESCRIPTION
**Pull request by davidbuckhurst**

### CODE REVIEW

File: app/models/job.rb [hunk 1]
```diff
@@ -40,6 +40,13 @@ def state_counts
   scope :queued, -> { where(state: "queued") }
   scope :running, -> { where(state: ["preparing", "running", "analyzing"] ) }
   delegate :queue_name, to: :job_group
+  
+  # Filter jobs for today's SLO targets
+  scope :core_hours_today, -> { day = 1.day.ago;
```
Comment (jrmhaig) [inline app/models/job.rb]: `:core_hours_yesterday` might be a better name.

File: app/services/builders/base.rb [hunk 2]
```diff
@@ -7,7 +7,8 @@ class Base
         tests:          { required: false, field_type: :array, description: "List of tests to be run", default_value: [""] },
         tests_per_job:  { required: false, field_type: :integer, description: "The number of tests to be run per job", default_value: "10" },
         jobs_per_queue: { required: false, field_type: :integer, description: "The number of jobs to create per queue (overrides tests_per_job)", default_value: nil },
-        retries:        { required: false, field_type: :integer, description: 'The number of automatic retry attempts', default_value: Chamber.env.maximum_auto_retries}
+        retries:        { required: false, field_type: :integer, description: 'The number of automatic retry attempts', default_value: Chamber.env.maximum_auto_retries},
+        job_timeout:    { required: false, field_type: :integer, description: 'Minutes a job can run before it is killed', default_value: 120}
```
Comment (Asimk21) [inline app/services/builders/base.rb]: Is it possible to set default value of job_timeout to nil and configure it either from runner or scheduler and ignore if not set and use default 5 hours value stuck_running_jobs_timeout: <%= 5.hours %>?

Comment (davidbuckhurst) [inline app/services/builders/base.rb]: Would we realistically ever want no timeout?

File: app/models/job.rb (hunk 1 above)
Comment (davidbuckhurst) [inline app/models/job.rb]: Yep, although I think this is a bit of a BBC-specific implementation here. Maybe something like: slo_core_period ?

**Pull request merged**